### PR TITLE
feat: CCIE-900 Automatically check whether Happy is up to date

### PR DESCRIPTION
**Pull request by dspadea**
- Add Release.Equal() to compare releases
- Add Function to fetch latest available version from hapi
- Do automated version check after command executes, and warn if Happy is not current
- Expose available-version command to allow on-demand fetching of current version number of Happy

### CODE REVIEW

File: cli/cmd/root.go [hunk 1]
```diff
@@ -107,5 +107,24 @@ func Execute() error {
...
+	outdated, cliVersion, latestAvailableVersion, err := IsHappyOutdated(cmd)
+
+	if err != nil {
+		log.Errorf("Error checking for latest available version number: %v", err)
+		return
+	}
+
+	if outdated {
+		log.Warnf("This copy of Happy CLI is not the latest availble. Please update at your earliest convenience.")
+		log.Warnf("CLI version: %s  Latest available version: %s", cliVersion.Version, latestAvailableVersion.Version)
```
Comment (jakeyheath) [inline cli/cmd/root.go:127]: Can we print the command used to update?

`brew upgrade happy`

Comment (dspadea) [reply cli/cmd/root.go:127]: Sure! Good idea!

Comment (dspadea) [reply cli/cmd/root.go:127]: Now reads:

```
[WARNING]: This copy of Happy CLI is not the latest available. CLI version: 999.12.34-SNAPSHOT-26b60f8  Latest available version: v0.64.0

[WARNING]: To update on Mac, run:  brew upgrade happy
```

File: cli/cmd/version.go [hunk 2]
```diff
@@ -22,3 +26,49 @@ var versionCmd = &cobra.Command{
...
+
+	api := hapi.MakeApiClient(happyConfig)
+	result := model.HealthResponse{}
+	err = api.GetParsed("/health", "", &result)
+	if err != nil {
+		return nil, err
+	}
+
+	return &util.Release{
+		Version: result.Version,
```
Comment (jakeyheath) [inline cli/cmd/version.go:61]: Also, just to confirm. The version that comes back from the API, this is the CLI version? Not the API version right?

Comment (jakeyheath) [reply cli/cmd/version.go:61]: jk, I just checked it out. These should be the same

Comment (dspadea) [reply cli/cmd/version.go:61]: Good point, and thanks for checking!

**Pull request merged**